From 6dac9ac289d6b9121a3683a6dbb69500408ed1d4 Mon Sep 17 00:00:00 2001
From: Xiaoyao Li <xiaoyao.li@intel.com>
Date: Tue, 31 Aug 2021 15:34:42 +0800
Subject: [PATCH 0486/1418] KVM: MMU: TDX: Pass base_gfn to kvm_x86_{unzap,
 set}_private_spte()

TDH.MEM.PAGE.{AUG, ADD} and TDH.MEM.RANGE.UNBLOCK require the gpa to be
page-level aligned.

Signed-off-by: Xiaoyao Li <xiaoyao.li@intel.com>
---
 arch/x86/kvm/mmu/mmu.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/arch/x86/kvm/mmu/mmu.c b/arch/x86/kvm/mmu/mmu.c
index b07a13aaf761..54584c78a923 100644
--- a/arch/x86/kvm/mmu/mmu.c
+++ b/arch/x86/kvm/mmu/mmu.c
@@ -3383,11 +3383,9 @@ static int __direct_map(struct kvm_vcpu *vcpu, gpa_t gpa, u32 error_code,
 			direct_pte_prefetch(vcpu, it.sptep);
 	} else if (!WARN_ON_ONCE(ret != RET_PF_FIXED)) {
 		if (is_zapped_pte)
-			static_call(kvm_x86_unzap_private_spte)(vcpu->kvm, gfn,
-								level);
+			static_call(kvm_x86_unzap_private_spte)(vcpu->kvm, base_gfn, level);
 		else
-			static_call(kvm_x86_set_private_spte)(vcpu, gfn, level,
-							      pfn);
+			static_call(kvm_x86_set_private_spte)(vcpu, base_gfn, level, pfn);
 	}
 
 	++vcpu->stat.pf_fixed;
-- 
2.31.1

